src/Xamarin.Android.Build.Tasks/Utilities/Aapt2Daemon.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks {
+	class Aapt2Daemon : IDisposable {
+		struct DaemonProcess {
+			public Process Process;
+			public object LockObject;
+		}
+		ConcurrentDictionary<int, DaemonProcess> daemons = new ConcurrentDictionary<int, DaemonProcess> ();
+
+		public string Aapt2 { get; private set; }
+
+		public string ToolName { get { return Path.GetFileName (Aapt2); }}
+
+		public static Aapt2Daemon GetInstance (IBuildEngine4 engine, string aapt2) {
+			Aapt2Daemon daemon = (Aapt2Daemon)engine.GetRegisteredTaskObject (typeof(Aapt2Daemon).FullName,  RegisteredTaskObjectLifetime.Build);
+			if (daemon == null)
+			{
+				daemon = new Aapt2Daemon (aapt2);
+				engine.RegisterTaskObject (typeof (Aapt2Daemon).FullName, daemon, RegisteredTaskObjectLifetime.Build, allowEarlyCollection: false);
+			}
+			return daemon;
+		}
+
+		public Aapt2Daemon (string aapt2)
+		{
+			Aapt2 = aapt2;
+		}
+
+		/*
+		`aapt2 daemon` allows commands to be passed via the StdIn Stream.
+		However each "argument" or "switch" needs to be on its own line.
+		so 
+			'--manifest AndroidManifest.xml'
+		will NOT work
+			`--manifest`
+			`AndroidManifest.xml`
+		does work.
+
+		This needs to be done for ALL arguments. 
+		*/
+		bool Run (DaemonProcess aapt2, IList<OutputLine> output, params string[] args) {
+			lock (aapt2.LockObject) {
+				StreamWriter writer = aapt2.Process.StandardInput;
+				Console.WriteLine ($"Writer = {writer.Encoding}");
+				foreach (var arg in args) {
+					writer.WriteLine (arg.Trim ());
+				}
+				writer.WriteLine ();
+				writer.Flush ();
+			}
```

Comment:
I think this lock might need to extend to the part reading from stderr, too?

Otherwise one thread might read another's standard error?